tools/jdbc/src/main/java/org/duckdb/DuckDBArrowResult.java

```diff
@@ -0,0 +1,66 @@
+package org.duckdb;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+@SuppressWarnings("rawtypes")
+public class DuckDBArrowResult {
+
+	private Class vector_root_class;
+	private Class schema_class;
+	private Class dictionary_class;
+
+	private Class root_allocator_class;
+	private Class buffer_allocator_class;
+
+	private Class c_schema_class;
+	private Class c_array_class;
+	private Class c_data_class;
+	private Class c_dictionary_class;
+
+	private Object allocator;
+
+	private DuckDBResultSet result_set;
+	private Object schema;
+
+	@SuppressWarnings("unchecked")
+	public DuckDBArrowResult(ResultSet result_set_p) throws Exception {
+		if (!(result_set_p instanceof DuckDBResultSet)) {
+			throw new SQLException("This only works with a DuckDB ResultSet");
+		}
+		result_set = (DuckDBResultSet) result_set_p;
+
+		vector_root_class = Class.forName("org.apache.arrow.vector.VectorSchemaRoot");
+		schema_class = Class.forName("org.apache.arrow.vector.types.pojo.Schema");
+		dictionary_class = Class.forName("org.apache.arrow.vector.dictionary.DictionaryProvider");
+
+		root_allocator_class = Class.forName("org.apache.arrow.memory.RootAllocator");
+		buffer_allocator_class = Class.forName("org.apache.arrow.memory.BufferAllocator");
+
+		c_schema_class = Class.forName("org.apache.arrow.c.ArrowSchema");
+		c_array_class = Class.forName("org.apache.arrow.c.ArrowArray");
+		c_data_class = Class.forName("org.apache.arrow.c.Data");
+		c_dictionary_class = Class.forName("org.apache.arrow.c.CDataDictionaryProvider");
+
+		allocator = root_allocator_class.getConstructor().newInstance();
+
+		Object c_schema = c_schema_class.getMethod("allocateNew", buffer_allocator_class).invoke(null, allocator);
+		result_set.arrowExportSchema((Long) c_schema_class.getMethod("memoryAddress").invoke(c_schema));
+		schema = c_data_class.getMethod("importSchema", buffer_allocator_class, c_schema_class, c_dictionary_class)
+				.invoke(null, allocator, c_schema, null);
+	}
+
+	@SuppressWarnings("unchecked")
+	public Object Fetch() throws Exception {
```

Comment:
Java methods should start with a lowercase letter

Reply:
thanks!